index.ts

```diff
@@ -180,6 +181,32 @@ export namespace log {
   }
 }
 
+/**
+ * The result of an operation, with a corresponding value and error type.
+ */
+export class Result<V, E> {
+  _value: Wrapped<V> | null
+  _error: Wrapped<E> | null
+
+  get isOk(): boolean {
```

Comment:
I think this is the best. It allows us the flexibility to change the implementation with a non-regrettable API.

Reply:
I was _so_ tempted to call `value` and `error` `unwrap` and `unwrapErr`. 😁